pip/req/req_file.py

```diff
@@ -1,137 +1,314 @@
+                # Remove after 7.0
                 finder.allow_unverified |= set([normalize_name(line).lower()])
+            elif opt == '-f' or opt == '--find-links':
+                # FIXME: it would be nice to keep track of the source
+                # of the find_links: support a find-links local path
+                # relative to a requirements file.
+                req_dir = os.path.dirname(os.path.abspath(filename))
+                relative_to_reqs_file = os.path.join(req_dir, value)
+                if os.path.exists(relative_to_reqs_file):
+                    value = relative_to_reqs_file
+                finder.find_links.append(value)
+
+        # ---------------------------------------------------------------------
+        elif linetype == IGNORE:
+            pass
+
+
+def parse_line(line):
+    if not line.startswith('-'):
+        # Split the requirement from the options.
+        if ' --' in line:
+            req, opts = line.split(' --', 1)
+            opts = parse_requirement_options('--%s' % opts)
         else:
-            comes_from = '-r %s (line %s)' % (filename, line_number)
-            if line.startswith(('-e', '--editable')):
-                editable = _remove_prefixes(line, '-e', '--editable')
-                req = InstallRequirement.from_editable(
-                    editable,
-                    comes_from=comes_from,
-                    default_vcs=options.default_vcs if options else None,
-                    isolated=options.isolated_mode if options else False,
-                )
+            req = line
+            opts = {}
+
+        return REQUIREMENT, (req, opts)
+
+    firstword, rest = partition_line(line)
+    # -------------------------------------------------------------------------
+    if firstword == '-e' or firstword == '--editable':
+        return REQUIREMENT_EDITABLE, rest
+
+    # -------------------------------------------------------------------------
+    if firstword == '-r' or firstword == '--requirement':
+        return REQUIREMENT_FILE, rest
+
+    # -------------------------------------------------------------------------
+    if firstword in parser_flags:
+        if rest:
+            msg = 'Option %r does not accept values.' % firstword
+            raise RequirementsFileParseError(msg)
+        return FLAG, firstword
+
+    # -------------------------------------------------------------------------
+    if firstword in parser_options:
+        if not rest:
+            msg = 'Option %r requires value.' % firstword
+            raise RequirementsFileParseError(msg)
+        return OPTION, (firstword, rest)
+
+    # -------------------------------------------------------------------------
+    if firstword in parser_compat:
+        return IGNORE, line
+
+
+def parse_requirement_options(args):
+    args = shlex.split(args)
+    opts, _ = _req_parser.parse_args(args)
+
+    if opts.install_options:
+        opts.install_options = flat_shlex_split(opts.install_options)
```

Comment:
Agreed - supplying ```--global-option='--author --author-email'``` should be an error rather than passing 2 arguments. At least that's how I believe the command line ```--global-option``` works (the docs aren't that clear) and it's what I think makes sense.